text/0000-default-field-values.md

```diff
@@ -0,0 +1,1685 @@
+- Feature Name: `default_field_values`
+- Start Date: 2018-12-17
+- RFC PR:
+- Rust Issue:
+
+# Summary
+[summary]: #summary
+
+Allow `struct` definitions and `enum` variants to provide default values for
+individual fields and thereby allowing those to be omitted from initializers.
+When deriving `Default`, the provided values will then be used. For example:
+
+```rust
+#[derive(Default)]
+struct ExprType {
+    expr: Box<Expr>,
+    ty: Box<Ty>,
+    attrs: Vec<Attribute> = Vec::new(),
+}
+```
+
+An attribute `#[default]`, usable on `enum` variants, is also introduced,
+thereby allowing enums to work with `#[derive(Default)]`.
+
+```rust
+#[derive(Default)]
```

Comment:
Could we use `#[derive(Default = Field)]` instead of the extra attribute? It has the advantages of not letting the programmer annotate multiple variants and fewer attribute, but does require typing the variant name twice.